packages/dashboard/src/components/DashboardGridLayout/DashboardGridLayoutWithAddCardPanel/DashboardGridLayoutWithAddCardPanel.tsx

```diff
@@ -222,8 +254,8 @@ export class DashboardGridLayoutWithAddCardPanel extends BaseComponent<
         newLayout.lg!.push(itemLayout);
       });
 
-      // for object comparision using Object.id method
-      if (Object.is(newLayout, this.state.layout)) {
+      // for object comparison using JSON.stringify method
+      if (JSON.stringify(newLayout) === JSON.stringify(this.state.layout)) {
```

Comment:
This is not a reliable method for object comparison because keys may be randomly ordered. https://stackoverflow.com/questions/15376185/is-it-fine-to-use-json-stringify-for-deep-comparisons-and-cloning